src/ReachSets/discretize.jl

```diff
@@ -720,3 +739,154 @@ function _discretize_interpolation_inhomog(δ, U0, U, X0, ϕ, Einit, Eψ0, A, si
+interval matrix.
+
+### Input
+
+- `𝑆`              -- a continuous system
+- `δ`              -- step size
+- `order`          -- order for Taylor-series approximation, ≥ 2
+- `set_operations` -- (optional, default: `"zonotope"`) set operations used for
+                      the discretized initial states and transformed inputs;
+                      choose among:
+
+    - `"lazy"`     -- use lazy convex hull for the initial states and lazy
+                      linear map for the inputs
+    - `"zonotope"` -- use concrete zonotope operations (linear map and Minkowski
+                      sum) and return zonotopes for both the initial states and
+                      the inputs of the discretized system
+
+### Output
+
+The discretized system. In particular:
+
+- if the input system is homogeneous, a `LinearDiscreteSystem` is returned,
+- otherwise a `ConstrainedLinearControlDiscreteSystem` is returned.
+
+## Algorithm
+
+See the first two lines of Algorithm 1 in [1].
+
+[1] M. Althoff, O. Stursberg, M. Buss. Reachability Analysis of Linear Systems
+with Uncertain Parameters and Inputs. CDC 2007.
+"""
+function discretize_interval_matrix(𝑆::InitialValueProblem, δ::Float64,
+                                    order::Int;
+                                    set_operations::String="zonotope")
+    @assert order > 1 "order must be > 1 but was $order"
+    N = Float64
+
+    # used to dispatch on the value of the set operation
+    set_ops = Val(Symbol(set_operations))
+
+    A, X0 = 𝑆.s.A, 𝑆.x0
+    ϕ = expm_overapproximation(A, δ, order)
+    F = correction_hull(A, δ, order)
+
+    Ω0_homog = _discretize_interval_matrix_homog(X0, ϕ, F, set_ops)
+
+    # early return for homogeneous systems
+    if inputdim(𝑆) == 0
+        return IVP(CLDS(ϕ, stateset(𝑆.s)), Ω0_homog)
+    end
+
+    U = inputset(𝑆)
+    U0 = next_set(U, 1)
+    n = size(A, 1)
+    linear_maps = Vector{LinearMap{N}}(undef, order > 2 ? 3 : 2)
+
+    IδW = 1/2 * δ^2 * A + 1/6 * δ^3 * A * A
+    @inbounds for i in 1:n
+        IδW[i, i] *= δ
+    end
+    linear_maps[1] = LinearMap(IδW, U0)
+
+    linear_maps[2] = LinearMap(_expm_remainder(A, δ, order; n=n) * δ, U0)
+
+    zero_interval = IntervalMatrices.Interval(zero(N), zero(N))
+    if order > 2
+        fac_i_plus_1 = 6
+        Ai = A * A
+        δ_iplus1 = δ^3
+        M_sum = IntervalMatrix(fill(zero_interval, size(A)))
+        @inbounds for i in 3:order
+            fac_i_plus_1 *= i+1
+            δ_iplus1 *= δ
+            Ai *= A
+            1/fac_i_plus_1 * δ_iplus1 * Ai
+        end
+        linear_maps[3] = LinearMap(M_sum, U0)
+    end
+
+    Ω0, Ud = _discretize_interval_matrix_inhomog(U, Ω0_homog, linear_maps,
+                                                 set_ops)
+
+    # create identity interval matrix
+    one_interval = IntervalMatrices.Interval(one(N), one(N))
+    B = IntervalMatrix(fill(zero_interval, (n, n)))
+    @inbounds for i in 1:n
+        B[i, i] = one_interval
```

Comment:
This calls for making a new function in `IntervalMatrices` for that purpose.

Reply:
https://github.com/JuliaReach/IntervalMatrices.jl/issues/49